### PR TITLE
Fixed Date picker getBestDateTimePattern bug

### PR DESCRIPTION
**Pull request by sjanic**
When inflating a layout with a DatePicker in Robolectric, we get an IllegalArgumentException with a bad character pattern (see issue https://github.com/robolectric/robolectric/issues/2024). 

This pull request fixes the bug based on the solution from jshvarts in the above mention issue, and includes a test that exposes the bug.

### CODE REVIEW

File: robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java [hunk 1]
```diff
@@ -15,8 +21,27 @@
     Build.VERSION_CODES.LOLLIPOP })
 public class ShadowICUTest {
 
-  @Test
-  public void getBestDateTimePattern_returnsReasonableValue() {
-    assertThat(ICU.getBestDateTimePattern("hm", null)).isEqualTo("h:mm a");
-  }
+    @Test
+    public void getBestDateTimePattern_returnsReasonableValue() {
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java]: Can you make it two indents like the rest of the project then I'll merge.

File: robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java [hunk 2]
```diff
@@ -15,8 +21,27 @@
     Build.VERSION_CODES.LOLLIPOP })
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java]: I think we should add LOLLIPOP_MR1 + M to the list of tests run.

**Comment by sjanic** [comment]
I've fixed the indent, and added LOLLIPOP_MR1 to the sdk. I don't know how to add the M sdk to the config attribute, which is why it's missing.

Please let me know, if you how to add the M sdk or if I have to do anything else. :)

File: robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java [hunk 3]
```diff
@@ -1,22 +1,47 @@
...
 import org.robolectric.annotation.Config;
+import org.robolectric.util.ActivityController;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
 @RunWith(TestRunners.MultiApiWithDefaults.class)
 @Config(sdk = {
-    Build.VERSION_CODES.LOLLIPOP })
+  Build.VERSION_CODES.LOLLIPOP,
+  Build.VERSION_CODES.LOLLIPOP_MR1})
```
Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java]: Add Build.VERSION_CODES.M here also

**Comment by jongerrish** [comment]
...and then merge in master,

File: robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java (hunk 3 above)
Comment (sjanic) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java]: I'm getting a compile error that M doesn't exist. The M sdk is installed on my computer (API 23), and I have tried to rerun install-dependencies.rb with the M sdk installed.

Do I need to do anything else, in order for Build.VERSION_CODES to generate the M entry in its class?

Comment (jongerrish) [inline robolectric/src/test/java/org/robolectric/shadows/ShadowICUTest.java]: AH, no my bad... we don't support M yet, I'm just adding it in the branch :-) This is great as is. Thanks.

**Pull request merged**